tests/e2e/utils/mock-requests.js

```diff
@@ -541,6 +541,23 @@ export default class MockRequests {
 		await this.fulfillAdsAccounts( [] );
 	}
 
+	/**
+	 * Mock Google Ads account as connected.
+	 *
+	 * @param {number} [id=12345]
+	 * @return {Promise<void>}
+	 */
+	async mockAdsSubAccountConnected( id = 123456 ) {
+		await this.fulfillAdsConnection( {
+			id,
+			sub_account: true,
+			created_timestamp: Date.now(),
+			currency: 'TWD',
+			symbol: 'NT$',
+			status: 'connected',
+		} );
+	}
```

Comment:
I think this won't be required if we accept the `args` in `mockAdsAccountConnected` and merge them with default args inside it.

@joemcgill Thoughts?

Reply:
I agree that we don't need a separate utility to mock a sub-account, but the bigger issue with this mock is that the `/wc/gla/ads/connection` endpoint rarely if ever returns the `sub_account` or `created_timestamp` values, from what I can tell.

https://github.com/woocommerce/google-listings-and-ads/blob/a3131d30776a4479692025f95abc9f6c66e70181/src/Ads/AccountService.php#L76-L94

Those values are only set during the setup process, otherwise they don't exist.

https://github.com/woocommerce/google-listings-and-ads/blob/02537c218b8c0be21c1d5d62f5bb39f3dce71474/src/Ads/AccountService.php#L152-L161

I can see that these values were originally an requirement for the logic in the `useFreeAdCredit()` hook, but something must have changed at some point. Rather than using that hook, lets use the same logic that is already in use to display the $500 ad credit in this card (explained in my other comment).

Reply:
👍